src/db/VectorDB.cpp

```diff
@@ -0,0 +1,77 @@
+#include <mw/db/VectorDB.h>
+#include "common/Database.h"
+#include "common/Serializable.h"
+
+VectorDB::VectorDB(const std::string& name, libmw::IDBWrapper* pDBWrapper, libmw::IDBBatch* pBatch)
+    : m_name("V" + name), m_pDatabase(std::make_unique<Database>(pDBWrapper, pBatch))
+{
+}
+
+VectorDB::~VectorDB() {}
+
+uint64_t VectorDB::Size() const
+{
+    auto pSize = m_pDatabase->Get<SerializableInt>(m_name, "Size");
+    if (pSize == nullptr) {
+        return 0;
+    }
+    return pSize->item->Get();
+}
+
+std::string VectorDB::IndexToString(uint64_t index)
+{
+    Serializer serializer;
+    serializer.Append(index);
+    return std::string(serializer.vec().begin(), serializer.vec().end());
+}
+
+std::unique_ptr<std::vector<uint8_t>> VectorDB::Get(uint64_t index) const
```

Comment:
I think we'll want to key by hash, not by index. If the consumer needs to lookup by index, they can easily retrieve the hash from the hash mmr, and then load from the DB via hash. It's unlikely we'll need to do many index lookups at all though. Most will actually be hash lookups.